akka-docs/rst/java/code/jdocs/akka/typed/IntroSpec.java

```diff
@@ -0,0 +1,229 @@
+import akka.typed.ExtensibleBehavior;
+import akka.typed.Signal;
+import akka.typed.javadsl.Actor;
+import akka.typed.javadsl.ActorContext;
+//#imports
+import java.util.ArrayList;
+import java.util.List;
+
+public class IntroSpec {
+
+  //#hello-world-actor
+  public static class HelloWorld {
+    public static final class Greet{
+      public final String whom;
+      public final ActorRef<Greeted> replyTo;
+
+      public Greet(String whom, ActorRef<Greeted> replyTo) {
+        this.whom = whom;
+        this.replyTo = replyTo;
+      }
+    }
+
+    public static final class Greeted {
+      public final String whom;
+
+      public Greeted(String whom) {
+        this.whom = whom;
+      }
+    }
+
+    public static final Behavior<Greet> greeter = Actor.stateless((ctx, msg) -> {
+      System.out.println("Hello " + msg.whom + "!");
+      msg.replyTo.tell(new Greeted(msg.whom));
+    });
+  }
+  //#hello-world-actor
+
+  //#chatroom-actor
+  public static class ChatRoom {
+    //#chatroom-protocol
+    static interface Command {}
+    public static final class GetSession implements Command {
+      public final String screenName;
+      public final ActorRef<SessionEvent> replyTo;
+      public GetSession(String screenName, ActorRef<SessionEvent> replyTo) {
+        this.screenName = screenName;
+        this.replyTo = replyTo;
+      }
+    }
+    //#chatroom-protocol
+    //#chatroom-behavior
+    private static final class PostSessionMessage implements Command {
+      public final String screenName;
+      public final String message;
+      public PostSessionMessage(String screenName, String message) {
+        this.screenName = screenName;
+        this.message = message;
+      }
+    }
+    //#chatroom-behavior
+    //#chatroom-protocol
+
+    static interface SessionEvent {}
+    public static final class SessionGranted implements SessionEvent {
+      public final ActorRef<PostMessage> handle;
+      public SessionGranted(ActorRef<PostMessage> handle) {
+        this.handle = handle;
+      }
+    }
+    public static final class SessionDenied implements SessionEvent {
+      public final String reason;
+      public SessionDenied(String reason) {
+        this.reason = reason;
+      }
+    }
+    public static final class MessagePosted implements SessionEvent {
+      public final String screenName;
+      public final String message;
+      public MessagePosted(String screenName, String message) {
+        this.screenName = screenName;
+        this.message = message;
+      }
+    }
+
+    public static final class PostMessage {
+      public final String message;
+      public PostMessage(String message) {
+        this.message = message;
+      }
+    }
+    //#chatroom-protocol
+    //#chatroom-behavior
+
+    public static Behavior<Command> chatRoom() {
+      return chatRoom(new ArrayList<ActorRef<SessionEvent>>());
+    }
+
+    private static Behavior<Command> chatRoom(List<ActorRef<SessionEvent>> sessions) {
```

Comment:
This style is all good (apart from that immutable collections are not commonly used in Java)